### DISCUSSION TITLE
Google/Microsoft floating timezones

### DISCUSSION THREAD

**Discussion by mischasigtermans** [spatie/icalendar-generator#139] [Ideas]
**✏️ Describe the bug**
In some cases you don't want to sync timezones to a calendar. Apple supports the use of [floating events](https://icalendar.org/CalDAV-Access-RFC-4791/7-3-date-and-floating-time.html), Google and Microsoft don't. For this case, `X-WR-TIMEZONE` should be set, with the corresponding local timezone (in my case `X-WR-TIMEZONE:Europe/Amsterdam`).

**✅ Expected behavior**
Would be nice to have a way to set the timezone on the Calendar.

```
Calendar::create()
   ->withoutTimezone('Europe/Amsterdam')
    ...
```

Resulting in

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:spatie/icalendar-generator
NAME:Calendar
X-WR-CALNAME:Calendar
X-WR-TIMEZONE:Europe/Amsterdam
BEGIN:VTIMEZONE
...
```

**Comment by rubenvanassche** [comment]
Hi @mischasigtermans looks like a cool feature, I would certainly accept a PR like tis!

**Reply by mischasigtermans** [reply]
I solved it with an appending property now.
